src/mock_alchemy/mocking.py

```diff
@@ -636,33 +636,51 @@ def _mutate_data(self, *args: Any, **kwargs: Any) -> Optional[int]:
             to_add = args[0]
             query_call = mock.call.query(type(to_add))
 
-            mocked_data = next(
-                iter(filter(lambda i: i[0] == [query_call], _mock_data)), None
-            )
+            mocked_data = next(iter(filter(lambda i: i[0] == [query_call], _mock_data)), None)
             if mocked_data:
                 mocked_data[1].append(to_add)
             else:
                 _mock_data.append(([query_call], [to_add]))
 
+            try:
+                execute_call = mock.call.execute(select(type(to_add)))
+
+                execute_mocked_data = next(
+                    iter(
+                        filter(
+                            lambda i: i[0] == [ExpressionMatcher(execute_call)],
+                            _mock_data,
+                        )
+                    ),
+                    None,
+                )
+                if execute_mocked_data:
+                    execute_mocked_data[1].append(to_add)
+                else:
+                    _mock_data.append(([execute_call], [to_add]))
+            except (TypeError, ArgumentError):
+                # TypeError indicates an old version of sqlalcemy that does not support
+                # executing select(table) statements.
+                # ArgumentError indicates a mocked table that cannot be selected, so
+                # cannot be mocked this way.
```

Comment:
For `ArgumentError ` shouldn't we inform the user that the mocked table could not be selected?

For `TypeError `, can we write this logic to a statement that is supported by multiple SqlAlchemy versions or do a check to verify the version supports executing select(table) statements rather relying on a exception to catch this case?